src/Tests/AzureClientEntryPointTests.cs

```diff
@@ -83,6 +85,338 @@ public async Task FromSnippet()
             Assert.IsNotNull(job);
         }
 
+        [TestMethod]
+        public async Task QIRSubmission()
+        {
+            var entryPointGenerator = Init("Workspace", new string[] { SNIPPETS.HelloQ });
+            var entryPoint = await entryPointGenerator.Generate("HelloQ", null, generateQir: true);
+
+            Assert.IsNotNull(entryPoint);
+            var job = await entryPoint.SubmitAsync(
+                new MockQIRSubmitter(new List<Argument>()),
+                new AzureSubmissionContext());
+            Assert.IsNotNull(job);
+        }
+
+        [TestMethod]
+        public async Task ValidParameterTypes()
+        {
+            var entryPointGenerator = Init("Workspace", new string[] { SNIPPETS.ValidParameterTypes });
+            var entryPoint = await entryPointGenerator.Generate("UseValidParameterTypes", null, generateQir: true);
+
+            Assert.IsNotNull(entryPoint);
+
+            var validArguments = new Dictionary<Argument, string>()
+            {
+                [new Argument("myBool", new ArgumentValue.Bool(true))] =  "\"true\"",
+                [new Argument("myDouble", new ArgumentValue.Double(1.2))] =  "\"1.2\"",
+                [new Argument("myInt", new ArgumentValue.Int(2))] =  "\"2\"",
+                [new Argument("myStr", new ArgumentValue.String("\"Hello\""))] = "\"Hello\"",
+                [new Argument("myPauli", new ArgumentValue.Pauli(Pauli.PauliX))] = "\"PauliX\"",
+                [new Argument("myResult", new ArgumentValue.Result(Result.One))] = "\"1\"",
+                [new Argument("innerInt", new ArgumentValue.Int(7))] = "\"7\"",
+                [new Argument("innerDouble", new ArgumentValue.Double(6.4))] = "\"6.4\""
+            };
+
+            var job = await entryPoint.SubmitAsync(
+                new MockQIRSubmitter(validArguments.Select(x => x.Key).ToList()),
+                new AzureSubmissionContext()
+                {
+                    InputParameters = validArguments.ToDictionary(x => x.Key.Name, x => x.Value)
+                });
+            Assert.IsNotNull(job);
+        }
+
+        [TestMethod]
+        public async Task ValidBoolParameter()
```

Comment:
There's a lot of repetition between this and the next several tests; would it be possible to deduplicate some? Thinking of a method like the following, perhaps?

```csharp
        internal async Task CheckValidParameter(
            string snippetSource,
            string operationName,
            string argumentName,
            ArgumentValue value,
            string valueAsExpr
        )
        {
            var entryPointGenerator = Init("Workspace", new [] { snippetSource });
            var entryPoint = await entryPointGenerator.Generate(operationName, null, generateQir: true);

            Assert.IsNotNull(entryPoint);
            var job = await entryPoint.SubmitAsync(
                new MockQIRSubmitter(new List<Argument>() { new Argument(argumentName, value) }),
                new AzureSubmissionContext()
                {
                    InputParameters = new Dictionary<string, string> { [argumentName] = valueAsExpr }
                });
            Assert.IsNotNull(job);
        }
```

Reply:
done